src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -450,6 +450,9 @@ bool emberAfOperationalCredentialsClusterAddNOCCallback(app::CommandHandler * co
     err = Server::GetInstance().GetFabricTable().Store(fabricIndex);
     VerifyOrExit(err == CHIP_NO_ERROR, nocResponse = ConvertToNOCResponseStatus(err));
 
+    // Notify the secure session of the new fabric.
+    commandObj->GetExchangeContext()->GetSessionHandle()->AsSecureSession()->NewFabric(commandObj->GetAccessingFabricIndex());
```

Comment:
Do we want to verify that this context was actually reached from a secure session?  Obviously the security model dictates it must be (and if not, I think this call will assert).  I'm just wondering if we want to double check.

Reply:
Alternatively, maybe this is more appropriately handled as part of #13711.

Reply:
My line of thought was, we are in commissioning adding NOCs, if we are doing that from an insecure session, don't we have bigger problems? That aside, if you have additional suggested checks that make sense here, or this could be better done from a different place, I'm open to that.

I do see other code in this file that does the same
`commandObj->GetExchangeContext()->GetSessionHandle()->AsSecureSession()`
string so if it's inappropriate here, it's probably inappropriate throughout this file.

At any rate, we need this to unblock ACL commissioning work, so it's better to get it in now and refactor it if we have improvements (I totally believe we probably do).

Reply:
Regarding #13711, yeah I had stronger checks in PASESession::NewFabric but they failed due to that problem. I removed them for the time being, until that's addressed. It is true though that this cannot affect CASE sessions, because I added a new member variable specifically for this special case, that seemed safer.

Reply:
It is true that this will fail if the session is somehow not a secure session.
```
SecureSession * Session::AsSecureSession()
{
    VerifyOrDie(GetSessionType() == SessionType::kSecure);
    return static_cast<SecureSession *>(this);
}
```

Reply:
One question though... do we allow commissioning that would come through here, over a CASE session? In that case, it would already be on a fabric, commissioning a new fabric I guess, and this code would not fail, but would not have an effect (since it's not a PASE session, the change would not be used). Thoughts?

Reply:
Checking the spec, seems this is possible, so what I did is fine: PASE "graduates" to the new fabric ID, but CASE does not.

Reply:
> Checking the spec, seems this is possible, so what I did is fine: PASE "graduates" to the new fabric ID, but CASE does not.

Yes, when @tcarmelveilleux and I worked through these questions, preserving the ability to AddNOC over CASE was very important.  And in the present form, @tcarmelveilleux 's text achieves this.  So yes, it is possible to get here form CASE, but only a PASE subject should acquire a fabric context from the AddNOC command.

However, we also need for this to only happen once.  AddNOC is only permissible once per fail-safe context, as otherwise cleanup logic is ambiguous on fail-safe timer expiration (if you have added multiple fabrics and the timer expires, which one will be deleted?).

And although not explicitly specified, that rule is probably also important for privileges here.  It does not seem proper for a PASE subject to acquire a different fabric association once it already has one.

Should we add a check for that then?  I don't think we should update the PASE subject fabric index unless it's currently the no-fabric index 0.

Reply:
Yeah I had that check (for only migrated off kUndefinedFabricIndex) while developing, but left it out for now. I can add it if we like it, sounds like we do.